### PR TITLE
update voting function

### PR DESCRIPTION
**Pull request by nampx-0860**

### CODE REVIEW

File: app/Http/Controllers/User/VoteController.php [hunk 1]
```diff
@@ -16,7 +18,14 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 6; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first();
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 22</br>Whitespace found at end of line

File: app/Http/Controllers/User/VoteController.php [hunk 2]
```diff
@@ -16,7 +18,14 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 6; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first(); 
+        $pollOption = Option::where('poll_id', $pollId)->get();
+        return response()
+            ->json(array(
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.ShortArraySyntax</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 25</br>"array()" function used. Use short array syntax, "[]", instead.

File: app/Http/Controllers/User/VoteController.php [hunk 3]
```diff
@@ -16,7 +19,14 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 6; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first();
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 23</br>Whitespace found at end of line

File: app/Http/Controllers/User/VoteController.php [hunk 4]
```diff
@@ -16,7 +19,14 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 6; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first(); 
+        $pollOption = Option::where('poll_id', $pollId)->get();
+        return response()
+            ->json(array(
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.ShortArraySyntax</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 26</br>"array()" function used. Use short array syntax, "[]", instead.

File: app/Http/Controllers/User/VoteController.php [hunk 5]
```diff
@@ -16,7 +19,15 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 6; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first();
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 23</br>Whitespace found at end of line

File: app/Http/Controllers/User/VoteController.php [hunk 6]
```diff
@@ -16,7 +19,15 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 5; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first();
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 23</br>Whitespace found at end of line

File: app/Http/Controllers/User/VoteController.php [hunk 7]
```diff
@@ -16,7 +19,19 @@ class VoteController extends Controller
...
+
+        $participantVote = ParticipantVote::with(['participant', 'option' => function($query){
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.KeySpecified</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 25</br>Key specified for array entry; first entry has no key

Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.SpaceAfterFunction</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 25</br>Expected 1 space after FUNCTION keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.SpaceBeforeBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 25</br>Expected 1 space before opening brace; found 0

File: app/Http/Controllers/User/VoteController.php [hunk 8]
```diff
@@ -16,7 +19,19 @@ class VoteController extends Controller
...
+            $query->where('poll_id', 5);
+        }])->get();
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 27</br>Closing bracket of array declaration must be on a new line

File: app/Http/Controllers/User/VoteController.php [hunk 9]
```diff
@@ -16,7 +19,19 @@ class VoteController extends Controller
...
+        $pollOption = Option::where('poll_id', $pollId)->get();
+        return response()
+            ->json([
+                'pollInfo' => $pollInfo,
+                'pollOption' => $pollOption,
+                'participantVote' => $participantVote
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 33</br>Each line in an array declaration must end in a comma

File: app/Http/Controllers/User/VoteController.php [hunk 10]
```diff
@@ -16,7 +19,21 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 5; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first();
+
+        $participantVote = ParticipantVote::with([
+            'participant',
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 26</br>Whitespace found at end of line

File: app/Http/Controllers/User/VoteController.php [hunk 11]
```diff
@@ -16,7 +19,21 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 5; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first();
+
+        $participantVote = ParticipantVote::with([
+            'participant', 
+            'option' => function($query){
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.KeySpecified</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 27</br>Key specified for array entry; first entry has no key

Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.SpaceAfterFunction</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 27</br>Expected 1 space after FUNCTION keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.SpaceBeforeBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 27</br>Expected 1 space before opening brace; found 0

File: app/Http/Controllers/User/VoteController.php [hunk 12]
```diff
@@ -16,7 +19,21 @@ class VoteController extends Controller
...
+                $query->where('poll_id', 5);
+            }])->get();
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 29</br>Closing bracket of array declaration must be on a new line

File: app/Http/Controllers/User/VoteController.php [hunk 13]
```diff
@@ -16,7 +19,22 @@ class VoteController extends Controller
      */
     public function index()
     {
-        //
+        $pollId = 5; //thay đổi id khi get link poll
+        $pollInfo = Poll::where('id', $pollId)->first();
+
+        $participantVote = ParticipantVote::with([
+            'participant',
+            'option' => function($query){
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.KeySpecified</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 27</br>Key specified for array entry; first entry has no key

Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.SpaceAfterFunction</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 27</br>Expected 1 space after FUNCTION keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/VoteController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.SpaceBeforeBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/User/VoteController.php</br>**Line**: 27</br>Expected 1 space before opening brace; found 0

**Comment by nampx-0860** [comment]
Peek 2019-04-01 14-00

File: resources/js/components/vote/tab/tabVote/FormInputInfo.js [hunk 14]
```diff
@@ -81,23 +86,89 @@ class FormInputInfo extends Component {
...
+                                                    </label>
+                                                </td>
+
+                                            </React.Fragment>
+                                        }.bind(this))
+                                    }
+
+                                </tr>
+                            </tbody>
+                        </table>
```
Comment (phamvanbk11) [inline resources/js/components/vote/tab/tabVote/FormInputInfo.js]: tách nhỏ ra thành nhiều component khác đi, ví dụ mỗi đoạn map là 1 component chẳng hạn, tách càng nhỏ càng tốt nhé

File: resources/js/components/vote/tab/tabVote/RowVote.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+                        </div>
+                    </th>
+                    <td className="table-name">{this.props.participantVote.participant.name}</td>
+                    <td className="table-name">{this.props.participantVote.participant.email}</td>
+                    {
+                        this.props.pollOption.map(function (option) {
+                            return <React.Fragment key={option.id}>
+                                <td>{option.id === this.props.participantVote.option_id ? <i className="fa fa-check"></i> : ''}</td>
+                            </React.Fragment>
+                        }.bind(this))
```
Comment (phamvanbk11) [inline resources/js/components/vote/tab/tabVote/RowVote.js]: đoạn này khai báo biến ở ngoài rồi gọi trong render nhé

File: resources/js/components/vote/tab/tabVote/TabVote.js [hunk 16]
```diff
@@ -8,18 +8,38 @@ class TabVote extends Component {
...
+        }
+    }
+    componentDidMount() {
+        axios.get(window.Laravel.baseUrl + '/api/vote')
+            .then(response => {
+                this.setState({
+                    pollInfo: response.data.pollInfo,
+                    pollOption: response.data.pollOption,
+                    participantVote: response.data.participantVote,
+                    pollId: response.data.pollId,
```
Comment (phamvanbk11) [inline resources/js/components/vote/tab/tabVote/TabVote.js]: tham khảo 
https://www.sitepoint.com/es6-destructuring-assignment/
để viết code ngắn gọn hơn

**Pull request merged**